### DISCUSSION TITLE
Pass Product Name Into Wix

### DISCUSSION THREAD

**Discussion by MaroisConsulting** [wixtoolset/issues#7221] [Questions]
I'm new to Wix, so please bear with me..

I'm going to be using one Wix installer for 3 different products. I have a Proof of Concept working now in Visual Studio 2022, but I'd like to be able to pass in name of the product at build time. I cannot seem to find any decent examples of how to do this. Can someone show me how this is done?

Thanks

**Comment by robmen** [comment]
I covered this in the Deployment Dojo, I think this episode https://youtube.com/live/g1PBKhq2oe8?feature=share

**Reply by MaroisConsulting** [reply]
Great thanks. I'll take a look

**Reply by MaroisConsulting** [reply]
Rob,

I'm watching the vid now, and it all makes sense. But I've got a problem I can't figure out...

Using VS2022 and Wix311

Wix Project:
```
<ProductInfo>MyApp</ProductInfo>
<VersionInfo>1.0.0.0</VersionInfo>
<DefineConstants>ProductInfo=$(ProductName);VersionInfo=$(Version)</DefineConstants>
```

Product:
```
<Product Id="*" 
			Name="$(var.ProductInfo)"
			Version="$(var.VersionInfo)"
			Language="1033" 
			Manufacturer="Acme Widgets" 
			UpgradeCode="{0C6D0563-C288-4217-90ED-904DCF799609}">

```
When I build from within VS I get
Error The Product/@Version attribute's value cannot be an empty string.	
Error The Product/@Version attribute was not found; it is required.
Error The Product/@Name attribute's value cannot be an empty string.


Command line
`msbuild -v:m WixDemo\WixDemo.sln -p:ProductName="My Installer" -p:Version=2.4.6.8888 `

When I build from the command line I get
`CNDL0150: Undefined preprocessor variable '$(var.ProductInfo)'.`

What am I doing wrong??

**Reply by BMurri** [reply]
In the project file, you create two MSBuild variables (ProductInfo & VersionInfo), neither one of which are consumed anywhere visible.
Then, you create and pass two Wix preprocessor user-defined variables (ProductInfo & VersionInfo), and each of those are set to the MSBuild/Environment variables ProductName & Version, respectively, neither of which have been assigned any value in the snippets you provided, which will result in the two WiX preprocessor variables having blank/empty values unless they were assigned values elsewhere.
